src/NServiceBus.Core/StartableEndpoint.cs

```diff
@@ -32,7 +32,7 @@ public async Task<IEndpointInstance> Start()
 
             AppDomain.CurrentDomain.SetPrincipalPolicy(PrincipalPolicy.WindowsPrincipal);
 
-            await receiveComponent.Initialize(containerComponent, recoverabilityComponent).ConfigureAwait(false);
+            await receiveComponent.PrepareToStart(containerComponent, recoverabilityComponent).ConfigureAwait(false);
```

Comment:
While I understand that this is a preparation stage and not yet receiving, the name with "Start" is confusing a bit.
There's a `Start` and there's a `PrepareToStart`.

Reply:
Yea, PrepareToStart is an anomaly, I have some ideas to remove it but it will require a TransportComponent first. I'd say it a step in the right direction